qcodes/dataset/data_set.py

```diff
@@ -756,6 +758,36 @@ def subscribe(self,
         subscriber.start()
         return subscriber_id
 
+    def subscribe_from_config(self, name: str) -> str:
+        """
+        Subscribe a subscriber defined in the `qcodesrc.json` config file to
+        the data of this `DataSet`. The definition can be found at
+        `subscription.subscribers`.
+
+        Args:
+            name: identifier of the subscriber. Equal to the key of the entry
+                in 'qcodesrc.json::subscription.subscribers'.
+        """
+        subscribers = qcodes.config.subscription.subscribers
+        try:
+            subscriber_info = getattr(subscribers, name)
+        except AttributeError:
+            keys = ','.join(subscribers.keys())
+            raise RuntimeError(
```

Comment:
could you please add a test for this exception? :)

Reply:
That was an impressive catch on your side: The dotdict that makes the config system possible does not throw an AttributeError here but a KeyError.

Reply:
it's not a catch, it's being (sometimes overly) "pedantic"

Reply:
great, thanks for adding the test! but why are we catching both types of exceptions?